Public/Src/FrontEnd/Nuget/NugetResolver.cs

```diff
@@ -62,12 +69,89 @@ public override async Task<bool> InitResolverAsync(IResolverSettings resolverSet
 
             m_owningModules = new Dictionary<ModuleId, Package>();
 
-            foreach (var package in maybePackages.Result)
+            foreach (var package in maybePackages.Result.Values.SelectMany(v => v))
             {
                 m_packages[package.Id] = package;
                 m_owningModules[package.ModuleId] = package;
             }
 
+            if (Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid ||
+                Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid)
+            {
+                var cgManfiestGenerator = new NugetCgManifestGenerator(Context);
+                string generatedCgManifest = cgManfiestGenerator.GenerateCgManifestForPackages(maybePackages.Result);
+                string existingCgManifest = "INVALID";
+
+                if ( !Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid &&

```

Comment:
The !Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid was used to ignore the Validation flag when the generate flag is set, since the generated file will always be valid

Reply:
Yeah, but maybe the intended semantics of `/validate /generate` should be "validate existing file then generate a new one".  Also, those two flags may point to entirely different files!  Therefore, I'd treat them completely separately, i.e.,: (1) first run validate if specified, then (2) run generate if specified.